spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/ResponseData.java

```diff
@@ -60,15 +72,37 @@ public ResponseData(ClientResponse response, RequestData requestData) {
 		this(response.statusCode(), response.headers().asHttpHeaders(), response.cookies(), requestData);
 	}
 
+	// Done this way to maintain backwards compatibility while allowing switching to raw
+	// HTTPStatus
+	// Will be removed in `4.x`
+	public ResponseData(RequestData requestData, ClientResponse response) {
+		this(response.headers().asHttpHeaders(), response.cookies(), requestData, response.rawStatusCode());
+	}
+
 	public ResponseData(ServerHttpResponse response, RequestData requestData) {
 		this(response.getStatusCode(), response.getHeaders(), response.getCookies(), requestData);
 	}
 
+	// Done this way to maintain backwards compatibility while allowing switching to raw
+	// HTTPStatus
+	// Will be removed in `4.x`
+	public ResponseData(RequestData requestData, ServerHttpResponse response) {
```

Comment:
Add deprecated to those marked "Will be removed in `4.x`".

Reply:
Done.